### DISCUSSION TITLE
Way to do Forms with Multi View(unmounting <Fields />)

### DISCUSSION THREAD

**Discussion by shafkathullah** [houseform/houseform#97] [Q&A] [answered]
Here [code link](https://stackblitz.com/edit/houseform-v1-example-html-forms-zrhgd7?file=App.tsx) I am trying to have 3 email fields `email_1`, `email_2` and `email_3`, but HouseForm seem to remove unmounted `Fields` from form altogether. Any ideas on how to implement such a scenario? Also I need to validate fields when `Next` button is pressed.

**Comment by joaom00** [comment, marked as answer]
We want to support this use case in #86. Would love to hear if you have any feedback on the potential API or features you want to see!

In the meantime, you need to wrap each step in a `Form` and when submit store the form values in a state and go to the next step.

Also, you can use the `preserveValue` prop to keep the field's values when unmount, but I think you won't need it in your case.

**Comment by shafkathullah** [comment]
ok, so if `preserveValue` preserves value on unmount, then why I don't see it when the form submits? Also a `Form` level validate function will be useful to call on `onClick` stepper `Next` button.

**Reply by crutchcorn** [reply]
If you're not seeing the value when the form submits, that's a bug. What version of HouseForm are you using?

As for the onClick stepper - that's a FormGroup (https://github.com/houseform/houseform/issues/86) feature request - we likely won't be able to get to that one right away

**Comment by shafkathullah** [comment]
No, I am not seeing the value on submit, only last value is being shown. Version is `1.10.0`. [code link](https://stackblitz.com/edit/houseform-v1-example-html-forms-zrhgd7?file=App.tsx)

**Reply by joaom00** [reply]
In your code, the `Field` is not being unmounted, so it is not preserving the values. you have two options:

- Create a field for each step:
```tsx
{page === 1 && <Field name={`email_${page}`}>{...}</Field>}
{page === 2 && <Field name={`email_${page}`}>{...}</Field>}
{page === 3 && <Field name={`email_${page}`}>{...}</Field>}
```

- Or use react's `key` prop to unmount:
```tsx
<Field key={`email_${page}`} name={`email_${page}`}>{...}</Field>
```

You can see working here https://stackblitz.com/edit/houseform-v1-example-html-forms-8k4b5t?file=App.tsx

**Reply by shafkathullah** [reply]
neat, i got re-rendering confused with mounting, this is cool, thanks!
